jwst/associations/lib/tests/test_diff.py

```diff
@@ -307,10 +306,28 @@ def test_fails(mismatched, standard=standard_asn):
     standard: Association
         The standard association.
     """
-    with pytest.raises(AssertionError):
+    with pytest.raises(AssertionError) as excinfo:
         left_asns = asn_diff.separate_products(mismatched)
         right_asns = asn_diff.separate_products(standard)
         asn_diff.compare_asn_lists(left_asns, right_asns)
+    assert "Products in left but not right:" in str(excinfo.value)
+    assert "Products in right but not left:" in str(excinfo.value)
```

Comment:
Ideally we only want to catch the exception from the line we expect to see it. Otherwise we might miss a bug if somehow that exception comes from a different line. Also the string comparison can probably be simplified with `match`.

```suggestion
    left_asns = asn_diff.separate_products(mismatched)
    right_asns = asn_diff.separate_products(standard)
    with pytest.raises(AssertionError, match="Products in .* but not .*"):
        asn_diff.compare_asn_lists(left_asns, right_asns)
```